src/client/common/process/interpreterActivation.ts

```diff
@@ -210,139 +235,64 @@ export class EnvironmentActivationService implements IEnvironmentActivationServi
             return this.activatedEnvVariablesCache.get(key);
         }
 
-        const promise = (async () => {
+        const customEnvVarsPromise = this.envVarsService.getEnvironmentVariables(resource);
+        // If this is a conda environment that supports conda run, then we don't need conda activation commands.
+        let activationCommandsPromise = this.getActivationCommands(resource, interpreter);
```

Comment:
Seems like this should be undefined (or Promise.resolve empty as default) and then in the if statement below if it doesn't support live streaming, then do this work?

Reply:
I want to get this as soon as possible, further down we have other awaits, and if that is `false`, then we've wasted time in not getting these commands.